adam/perception/perception_graph.py

```diff
@@ -510,9 +544,7 @@ class PerceptionGraphPatternMatching:
     pattern: PerceptionGraphPattern = attrib(
         validator=instance_of(PerceptionGraphPattern)
     )
-    graph_to_match_against: PerceptionGraph = attrib(
-        validator=instance_of(PerceptionGraph)
-    )
+    graph_to_match_against: TargetGraphT = attrib(validator=instance_of(TargetGraphT))
```

Comment:
You can't provide type variables to `instance_of`; the best you can do is test it is one of the two possible types.